doc/users/next_whats_new/datestr2num_list_with_default_date.rst

```diff
@@ -0,0 +1,12 @@
+Bug fixed in ``dates.datestr2num`` when processing list with default date
```

Comment:
This doesn't really need a What's new entry. Just give your PR a reasonably descriptive title.

Reply:
OK! I wasn't sure where the limit was. I'll change the title (the focus was really on test coverage, the bug fix was more an outcome of that).